src/core/agent/services/connectionService.ts

```diff
@@ -310,45 +308,38 @@ class ConnectionService extends AgentService {
       await this.credentialStorage.getCredentialMetadataByConnectionId(
         connectionId
       );
-    const credentialStatistics: {
-      [key: string]: { name: string; count: number };
-    } = {};
+    const credentialTypes: { [key: string]: string } = {};
```

Comment:
I would also prefer if we can just store the credential type in the IPEX message. Each message should have some reference to a schema (either in the message, or in the ACDC embedded) and we can get the name by schema using signify.

We should be able to do completely remove the call to `getCredentialMetadataByConnectionId` here, and just rely on the IPEX messages. (I think we can remove `getCredentialMetadataByConnectionId` from the repo, as it's only used here)